cypress/e2e/switchingFeedback.cy.ts

```diff
@@ -0,0 +1,128 @@
+import SurveyFeedback from "services/entities/SurveyFeedback"
+import hashedPassword from "../fixtures/hashedPassword"
+import TestCases from "../fixtures/switchingFeedbackTestData"
+
+describe("Switching Bichard Version Feedback Form", () => {
+  const expectedUserId = 0
+
+  before(() => {
+    cy.intercept("GET", "/bichard-ui/*", {
+      statusCode: 200,
+      body: {
+        name: "Dummy response"
+      }
+    })
+  })
+
+  after(() => {
+    cy.task("clearAllFeedbacksFromDatabase")
+  })
+
+  beforeEach(() => {
+    cy.viewport(1280, 720)
+    cy.task("clearAllFeedbacksFromDatabase")
+    cy.task("clearUsers")
+
+    cy.task("insertUsers", {
+      users: [
+        {
+          id: expectedUserId,
+          username: "Bichard01",
+          visibleForces: ["01"],
+          forenames: "Bichard Test User",
+          surname: "01",
+          email: "bichard01@example.com",
+          password: hashedPassword
+        }
+      ],
+      userGroups: ["B7NewUI_grp", "B7Supervisor_grp"]
+    })
+    cy.login("bichard01@example.com", "password")
+  })
+
+  it("Should access the switching feedback form with first question visible when user clicks 'Switch to old Bichard'", () => {
+    cy.visit("/bichard")
+    cy.contains("button", "Switch to old Bichard").click()
+    cy.get("a").contains("Back")
+    cy.get("button").contains("Skip feedback")
+    cy.get("h1").contains("Share your feedback")
+    cy.get("p")
+      .contains(
+        "You have selected to revert back to old Bichard. What was the reason for doing so? Can you please select the appropriate option. And outline the problem that occurred below so that we can best understand."
+      )
+      .should("exist")
+    cy.get("h3").contains("Why have you decided to switch version of Bichard?")
+    cy.get("h5").contains("Select one of the below option")
+    cy.get("label").contains(
+      "I have found an issue(s) when using the new version of Bichard which is blocking me from completing my task."
+    )
+    cy.get("label").contains("I prefer working in the old version, and I dislike working in the new version.")
+    cy.get("label").contains("Other (please specify)")
+  })
+
+  TestCases.forEach(([testName, testInput]) => {
```

Comment:
Wow, this test matrix is complex, but I guess makes sense and reduces a lot of code, the test data is fairly readable!